React/Modules/RCTUIManager.m

```diff
@@ -224,6 +224,8 @@ @implementation RCTUIManager
   NSDictionary *_componentDataByName;
 
   NSMutableSet<id<RCTComponent>> *_bridgeTransactionListeners;
+  NSMutableSet<id<RCTUIManagerObserver>> *_uiManagerObservers;
```

Comment:
This should hold weak references -- otherwise I think there is a cycle from the animation module -> bridge -> UIManager.

You might be able to use `[NSPointerArray weakObjectsPointerArray]` or an NSArray of `[NSValue valueWithNonretainedObjectValue:]` but be very careful especially around the semantics of what happens when the values are dealloc'd and whether the references are properly zeroed.